wgpu-hal/src/vulkan/instance.rs

```diff
@@ -236,48 +252,67 @@ impl super::Instance {
     /// - `extensions` must be a superset of `required_extensions()` and must be created from the
     ///   same entry, driver_api_version and flags.
     /// - `android_sdk_version` is ignored and can be `0` for all platforms besides Android
+    ///
+    /// If `debug_utils_user_data` is `Some`, then the validation layer is
+    /// available, so create a [`vk::DebugUtilsMessengerExt`].
     #[allow(clippy::too_many_arguments)]
     pub unsafe fn from_raw(
         entry: ash::Entry,
         raw_instance: ash::Instance,
         driver_api_version: u32,
         android_sdk_version: u32,
+        debug_utils_user_data: Option<super::DebugUtilsMessengerUserData>,
         extensions: Vec<&'static CStr>,
         flags: crate::InstanceFlags,
         has_nv_optimus: bool,
         drop_guard: Option<crate::DropGuard>,
     ) -> Result<Self, crate::InstanceError> {
         log::info!("Instance version: 0x{:x}", driver_api_version);
 
-        let debug_utils = if extensions.contains(&ext::DebugUtils::name()) {
-            log::info!("Enabling debug utils");
-            let extension = ext::DebugUtils::new(&entry, &raw_instance);
-            // having ERROR unconditionally because Vk doesn't like empty flags
-            let mut severity = vk::DebugUtilsMessageSeverityFlagsEXT::ERROR;
-            if log::max_level() >= log::LevelFilter::Debug {
-                severity |= vk::DebugUtilsMessageSeverityFlagsEXT::VERBOSE;
-            }
-            if log::max_level() >= log::LevelFilter::Info {
-                severity |= vk::DebugUtilsMessageSeverityFlagsEXT::INFO;
-            }
-            if log::max_level() >= log::LevelFilter::Warn {
-                severity |= vk::DebugUtilsMessageSeverityFlagsEXT::WARNING;
+        eprintln!(
+            "JIMB: Instance::from_raw(debug_utils_user_data = {:?})",
+            debug_utils_user_data
+        );
+        let debug_utils = if let Some(debug_callback_user_data) = debug_utils_user_data {
+            if extensions.contains(&ext::DebugUtils::name()) {
```

Comment:
Would be nice to have an info level print about why debug callbacks are disabled.